presto-tests/src/test/java/com/facebook/presto/server/TestQueryResource.java

```diff
@@ -78,9 +79,9 @@ public void testGetQueryInfos()
         runToCompletion("SELECT 1");
         runToCompletion("SELECT 2");
         runToCompletion("SELECT x FROM y");
-        runToFirstResult("SELECT * from tpch.sf100.orders");
-        runToFirstResult("SELECT * from tpch.sf100.orders");
-        runToFirstResult("SELECT * from tpch.sf100.orders");
+        runToFirstResult("SELECT * from tpch.sf100.orders limit 1");
```

Comment:
Why introduce the limit?  The purpose of selecting from `tpch.sf100.orders` is that this query is in running state for the checks below, however once we introduce the limit it seems it will actually be completed quickly?

Reply:
To check the returned order on lines 99-101.
I can change limit to 1B+1/2/3, if that will help.
But I thought that "runToFirstResult" will actually not finish the queries or was my assumption wrong?
If we rely on a timing of a query might finish or not, then this test should be flaky.
In fact, this test never works on my mac - the queries fail, which should not fail, so I'm kind of "blind" working with it.

Reply:
It will be impossible for `select * from tpch.sf100.orders` to finish in a platform dependent way, because:

* A query will not transition to finished state until the last page of data is retrieved from the client.
* `runToFirstResult` fetches the first page of data and then stops.
* There's many thousands or millions of pages remaining to be fetched, and it seems impossible to set a fetch size large enough to fetch in one shot in a unit test.

I think this should work just fine without the limit?  It will still be running, and the ordering should be unchanged.

Reply:
> I think this should work just fine without the limit? It will still be running, and the ordering should be unchanged.

I added ordering to we have statements differ from each other and we can check that they are return in a particular order we expect.
I'll let large limit then.
It works now, so I assume we should not be afraid of these queries finishing before we want them to finish.
Alternatively, I can add different comment to each of them, which is probably better.

Reply:
Oh I see.  Alternatively, you could also use different scale factors, such as `sf100`, `sf101`, and `sf102`.  The order of magnitude in each table still ensures they're running and not finished, but also differentiates them.